ClosedXML/Excel/CalcEngine/Expression.cs

```diff
@@ -509,10 +509,14 @@ public override object Evaluate()
 
         public IEnumerator GetEnumerator()
         {
-            if (_value is string)
-                return new[] { (string)_value }.GetEnumerator();
+            if (_value is string s)
+                yield return s;
 
-            return (_value as IEnumerable).GetEnumerator();
+            if (_value is IEnumerable ie)
+            {
+                foreach (var o in ie)
+                    yield return o;
+            }
```

Comment:
Maybe if `_value` is not `string` or `IEnumerable`, then we should just `yield return _value`?

Reply:
Yes, it makes sense.